### PR TITLE
replicate has all possible options and correct warnings

### PR DESCRIPTION
**Pull request by abuts**
Addresses #1495. 

These changes addresses small inconvenience in `replicate` providing common interface and common behaviour for `sqw` and `dnd` objects

### CODE REVIEW

File: horace_core/sqw/@sqw/sqw.m [hunk 1]
```diff
@@ -390,6 +390,8 @@ function  save_xye(obj,varargin)
         function npix = get.num_pixels(obj)
             npix = obj.pix_.num_pixels;
         end
+        % if sqw object has actual pixels or pixelles object
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@sqw/sqw.m:393]:
```suggestion
        % if sqw object has actual pixels or pixels object
```
avoid confusion that there might be boy pixels and girl pixels

File: horace_core/sqw/@DnDBase/replicate.m [hunk 2]
```diff
@@ -18,11 +18,16 @@
 %           - The data is expanded along the plot axes of wref that are
 %           integration axes of win.
 %           - The annotations etc. are taken from the reference dataset.
+% Optional:
+% '-set_pix' -- if provided, wref object should be sqw objects with pixels.
+%               in this case result would be sqw object(s) with pixels set
+%               to reprouce replicated image, defined by input dnd object.
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/replicate.m:24]:
```suggestion
%               to reproduce replicated image, defined by input dnd object.
```
You have said both that wref should be sqw and that input object is dnd. Something is inconsistent. Also as the current class is dndbase, checking if this is appropriate for sqw input/output

File: horace_core/sqw/@DnDBase/replicate.m [hunk 3]
```diff
@@ -18,11 +18,16 @@
...
 %
 % Output:
 % -------
 %   wout    Output dataset object (or array of objects). It is dnd object
-%           with the same dimensionality as wref.
+%           with the same dimensionality as wref or sqw object with the
+%           same dimensionality as input dnd object and
```
Comment (cmarooney-stfc) [inline horace_core/sqw/@DnDBase/replicate.m:30]: comment stopped short without completion

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/@sqw/sqw.m (hunk 1 above)
Comment (abuts) [reply horace_core/sqw/@sqw/sqw.m:393]: hm -- pixel-less - no pixels are present. "Pixelles" team may higack useful name but I would not even bother thinking about them. `pixels object` seems is something different from pixelles

**Pull request merged**